rio_toa/reflectance.py

```diff
@@ -56,15 +56,21 @@ def reflectance(img, MR, AR, E, src_nodata=0):
     input_shape = img.shape
 
     if len(input_shape) > 2:
-        img = np.rollaxis(img, 0, len(input_shape))
 
-    rf = (
-        ((MR * img.astype(np.float32)) + AR) / np.sin(np.deg2rad(E))
-        )
+        img = np.rollaxis(img, 0, len(input_shape))
 
+    rf = ((MR * img.astype(np.float32)) + AR) / np.sin(np.deg2rad(E))
     rf[img == src_nodata] = 0.0
 
-    return rf.reshape(input_shape)
```

Comment:
@dnomadb found a tiny bug post-rollaxis when handling multiple bands. Simply reshaping the array jumbles the values and produces the following *phish* tif
![jumbled_tiny](https://cloud.githubusercontent.com/assets/4450007/17008132/67b64b80-4ea3-11e6-9a23-1cb50b24ecdf.png)
instead of:
![pretty_tiny](https://cloud.githubusercontent.com/assets/4450007/17008151/93e72058-4ea3-11e6-9f6f-dc0d76330839.png)





```
>>> m = np.ones((3,2,4))
>>> x = [0.5, 2, 4]
>>> np.rollaxis(m, 0,3)*x
array([[[ 0.5,  2. ,  4. ],
        [ 0.5,  2. ,  4. ],
        [ 0.5,  2. ,  4. ],
        [ 0.5,  2. ,  4. ]],
       [[ 0.5,  2. ,  4. ],
        [ 0.5,  2. ,  4. ],
        [ 0.5,  2. ,  4. ],
        [ 0.5,  2. ,  4. ]]])

>>> (np.rollaxis(m, 0,3)*x).reshape((3,2,4))
array([[[ 0.5,  2. ,  4. ,  0.5],
        [ 2. ,  4. ,  0.5,  2. ]],
       [[ 4. ,  0.5,  2. ,  4. ],
        [ 0.5,  2. ,  4. ,  0.5]],
       [[ 2. ,  4. ,  0.5,  2. ],
        [ 4. ,  0.5,  2. ,  4. ]]])

>>> m * np.array(x)[:,None, None]
array([[[ 0.5,  0.5,  0.5,  0.5],
        [ 0.5,  0.5,  0.5,  0.5]],
       [[ 2. ,  2. ,  2. ,  2. ],
        [ 2. ,  2. ,  2. ,  2. ]],
       [[ 4. ,  4. ,  4. ,  4. ],
        [ 4. ,  4. ,  4. ,  4. ]]])

>>> np.rollaxis( np.rollaxis(m, 0,3)*x, 2,0)
array([[[ 0.5,  0.5,  0.5,  0.5],
        [ 0.5,  0.5,  0.5,  0.5]],
       [[ 2. ,  2. ,  2. ,  2. ],
        [ 2. ,  2. ,  2. ,  2. ]],
       [[ 4. ,  4. ,  4. ,  4. ],
        [ 4. ,  4. ,  4. ,  4. ]]])
```